### PR TITLE
Add an option to fixup vim popup with conceal problems

### PR DESCRIPTION
**Pull request by hrsh7th**
In vim, the conceal will cause the rendering problem with popup window.

I think it should be fixed in the vim side but I heard it would be difficult to fix.

This PR is adding the workaround for this problem.

### CODE REVIEW

File: autoload/vital/_lsp/VS/LSP/MarkupContent.vim [hunk 1]
```diff
@@ -36,27 +36,31 @@ function! s:normalize(markup_content, ...) abort
...
-
-  let l:string = substitute(l:string, "\\%(\\s\\|\n\\)*```\\s*\\(\\w\\+\\)\\%(\\s\\|\n\\)\\+", "\n\n```\\1 ", 'g')
-  let l:string = substitute(l:string, "\\%(\\s\\|\n\\)\\+```\\%(\\s*\\%(\\%$\\|\n\\)\\)\\+", " ```\n\n", 'g')
+function! s:_format(string, compact) abort
+  let l:string = a:string
+  if a:compact
+    let l:string = substitute(l:string, "\\%(\\s\\|\n\\)*```\\s*\\(\\w\\+\\)\\%(\\s\\|\n\\)\\+", "\n\n```\\1 ", 'g')
+    let l:string = substitute(l:string, "\\%(\\s\\|\n\\)\\+```\\%(\\s*\\%(\\%$\\|\n\\)\\)\\+", " ```\n\n", 'g')
+  else
+    let l:string = substitute(l:string, "```\n\\zs\\%(\\s\\|\n\\)\\+", "", 'g')
```
[the remark this answers is not in the document]
Comment (hrsh7th) [reply autoload/vital/_lsp/VS/LSP/MarkupContent.vim:60]: It's needed to use `\n` in regex pattern.

**Comment by hrsh7th** [comment]
The neovim tests is failing from anothor reason.

**Review by mattn — APPROVED** [review]

**Comment by mattn** [comment]
Thank you!

**Comment by mattn** [comment]
image

👍

**Pull request merged**